scripts/ci/build_emscripten.sh

```diff
@@ -34,55 +34,96 @@
 
 set -ev
 
-if test -z "$1"; then
-	BUILD_DIR="emscripten_build"
-else
-	BUILD_DIR="$1"
-fi
+SCRIPT_DIR="$(realpath "$(dirname "$0")/..")"
+# shellcheck disable=SC1091
+source "${SCRIPT_DIR}/common.sh"
 
-WORKSPACE=/root/project
+function build() {
+    local build_dir="$1"
+    local prerelease_source="${2:-ci}"
 
-cd "$WORKSPACE"
+    cd /root/project
 
-# shellcheck disable=SC2166
-if [[ "$CIRCLE_BRANCH" = release || -n "$CIRCLE_TAG" || -n "$FORCE_RELEASE" || "$(git tag --points-at HEAD 2>/dev/null)" == v* ]]
-then
-	echo -n >prerelease.txt
-else
-	# Use last commit date rather than build date to avoid ending up with builds for
-	# different platforms having different version strings (and therefore producing different bytecode)
-	# if the CI is triggered just before midnight.
-	TZ=UTC git show --quiet --date="format-local:%Y.%-m.%-d" --format="ci.%cd" >prerelease.txt
-fi
-if [ -n "$CIRCLE_SHA1" ]
-then
-	echo -n "$CIRCLE_SHA1" >commit_hash.txt
-fi
+    # shellcheck disable=SC2166
+    if [[ "$CIRCLE_BRANCH" = release || -n "$CIRCLE_TAG" || -n "$FORCE_RELEASE" || "$(git tag --points-at HEAD 2>/dev/null)" == v* ]]
```

Comment:
Just noticed that this is missing the check that `prerelease_source` is not set. You only changed the one in the other script.

Reply:
Oh yeah, missed that one. I will fix it.

Reply:
Actually, on second thought, for this script, it is not necessary to check if `prerelease_source` is set because the value is ignored in that case, and it has a default value, and it will always be set.

Reply:
>  it is not necessary to check if `prerelease_source` is set because the value is ignored in case of a release

Yeah, that was the point - I wanted it to report an error, not ignore it. I.e. the caller should not use `--prerelease-source` option when building a release.

But now that I look at it closer, you're right, we need to ignore this after all. I was assuming it was up to the caller to decide if he's building a release and that's the case with `$FORCE_RELEASE`. But I just realized that with `$CIRCLE_TAG` the job is the same and it only depends on whether we happen to be building on a tag or not.

In that case let's leave it here as is and let's remove this error from the other script too.